### PR TITLE
Remove error from `AnnotatedQueue.update_info`, deprecate `safe_update_info`

### PR DESCRIPTION
**Pull request by albi3ro**
Blocked by #3061 

A while back I created `safe_update_info` to be `update_info` without the `QueuingError` if the object wasn't in the queue.

For example:
```
q = AnnotatedQueue()
q.update_info(qml.PauliZ(0), key="value")
```
currently raises a `QueuingError`, while
```
q.safe_update_info(qml.PauliZ(0), key="value")
```
would not.

I don't see any reason to keep the error-raising version around, as it was not used anywhere.

So this removes the error raised by `update_info` and deprecates `safe_update_info`.

[sc-26065]

### CODE REVIEW

File: tests/test_queuing.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 import contextlib
 
 import pytest
+from torch import qint8
```
Comment (albi3ro) [inline tests/test_queuing.py:20]:
```suggestion
```

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -53,6 +53,8 @@
       is now only available through `QueuingManager`.
    - `AnnotatedQueue` and its children no longer have the private `_append`, `_remove`, `_update_info`, `_safe_update_info`,
       and `_get_info` methods. The public analogues should be used instead.
+   - `QueuingManager.safe_update_info` and `AnnotatedQueue.safe_update_info` are removed.  Their functionality is moved to
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:56]:
```suggestion
   - `QueuingManager.safe_update_info` and `AnnotatedQueue.safe_update_info` are deprecated.  Their functionality is moved to
```

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -62,6 +64,9 @@
  * `QueuingContext` is renamed `QueuingManager`.
   [(#3061)](https://github.com/PennyLaneAI/pennylane/pull/3061)
 
+ * `QueuingManager.safe_update_info` and `AnnotatedQueue.safe_update_info` are removed. Instead, `update_info` no longer raises errors
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:67]:
```suggestion
 * `QueuingManager.safe_update_info` and `AnnotatedQueue.safe_update_info` are deprecated. Instead, `update_info` no longer raises errors
```

**Review by AlbertMitjans — APPROVED** [review]
Nothing to change. Good job!! 🚀

**Pull request merged**